### PR TITLE
Enables HTTPs

### PR DESCRIPTION
**Pull request by eaceto**
Only if Basic Auth is enabled

### CODE REVIEW

File: bin/www [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 var app = require('../app');
 var http = require('http');
+var https = require('https');
```
Comment (nfmelendez) [inline bin/www]: use const

File: bin/www [hunk 2]
```diff
@@ -6,6 +6,8 @@
...
+var forge = require('node-forge');
```
Comment (nfmelendez) [inline bin/www]: use const

File: bin/www [hunk 3]
```diff
@@ -96,3 +114,87 @@ function onListening() {
     : 'port ' + addr.port;
   //console.log('Listening on ' + bind);
 }
+
+function createCertificate(keys, port) {
+  const pki = forge.pki;
+  // generate a keypair and create an X.509v3 certificate
+  var cert = pki.createCertificate();
```
Comment (nfmelendez) [inline bin/www]: use const

File: bin/www [hunk 4]
```diff
@@ -96,3 +114,87 @@ function onListening() {
...
+  cert.publicKey = keys.publicKey;
+
+  cert.serialNumber = '01';
+  cert.validity.notBefore = new Date();
+  cert.validity.notAfter = new Date();
+  cert.validity.notAfter.setFullYear(cert.validity.notBefore.getFullYear() + 10);
+  var attrs = [{
```
Comment (nfmelendez) [inline bin/www]: use const

File: bin/www [hunk 5]
```diff
@@ -24,10 +26,26 @@ app.set('httpUser', httpUser);
...
-var server = http.createServer(app);
+var server;
+
+if (httpUser !== undefined && httpPassword !== undefined && httpUser.length > 0 && httpPassword.length > 0) {
+  const pki = forge.pki;
+  const rsa = forge.pki.rsa;
+  const keys = rsa.generateKeyPair(2048);
+  const certificate = createCertificate(keys, port);
+
+  var server = https.createServer({
```
Comment (nfmelendez) [inline bin/www]: var defined in line 34

**Review by nfmelendez — COMMENTED** [review]
Small details

File: bin/www [hunk 6]
```diff
@@ -90,9 +108,93 @@ function onError(error) {
...
+      type: 7, // IP
+      ip: '127.0.0.1'
+    }]
+  }, {
+    name: 'subjectKeyIdentifier'
+  }]);
+  // self-sign certificate
+  cert.sign(keys.privateKey);
+  return cert
+}
```
Comment (mkiisoft) [inline bin/www]: break line? lint/analyzers normally complain.

**Review by mkiisoft — APPROVED** [review]

**Pull request merged**